File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Sae/AlibabaCloud/SDK/ROS/CDK/Sae/SlbBindingProps.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Sae
{
    #pragma warning disable CS8618

    /// <summary>Properties for defining a `ALIYUN::SAE::SlbBinding`.</summary>
    [JsiiByValue(fqn: "@alicloud/ros-cdk-sae.SlbBindingProps")]
    public class SlbBindingProps : AlibabaCloud.SDK.ROS.CDK.Sae.ISlbBindingProps
    {
        /// <summary>Property appId: Successful application deployment target ID.</summary>
        [JsiiProperty(name: "appId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        public object AppId
        {
            get;
            set;
        }

        /// <summary>Property internet: Binding public SLB.</summary>
        /// <remarks>
        /// For example: [{ "port": 80, "targetPort": 8080, "protocol": "TCP"}], shows a container port 8080 through port 80 slb exposed service, the protocol is TCP, the blank is ignored.
        /// </remarks>
        [JsiiOptional]
        [JsiiProperty(name: "internet", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? Internet
        {
            get;
            set;
        }

        /// <summary>Property internetSlbId: Use SLB purchased specified, currently only supports non-shared examples.</summary>
        [JsiiOptional]
        [JsiiProperty(name: "internetSlbId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? InternetSlbId
        {
            get;
            set;
        }

        /// <summary>Property intranet: Bind private SLB.</summary>
        /// <remarks>
        /// For example: [{ "port": 80, "targetPort": 8080, "protocol": "TCP"}], shows a container port 8080 through port 80 slb exposed service, the protocol is TCP, the blank is ignored.
        /// </remarks>
        [JsiiOptional]
        [JsiiProperty(name: "intranet", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? Intranet
        {
            get;
            set;
        }

        /// <summary>Property intranetSlbId: Use SLB purchased specified, currently only supports non-shared examples.</summary>
        [JsiiOptional]
        [JsiiProperty(name: "intranetSlbId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}", isOptional: true)]
        public object? IntranetSlbId
        {
            get;
            set;
        }
    }
}

File: multiple-languages/go/alicloudroscdksae/SlbBindingProps.go
package alicloudroscdksae


// Properties for defining a `SlbBinding`.
//
// See https://www.alibabacloud.com/help/ros/developer-reference/aliyun-sae-slbbinding
type SlbBindingProps struct {
	// Property appId: Successful application deployment target ID.
	AppId interface{} `field:"required" json:"appId" yaml:"appId"`
	// Property internet: Binding public SLB.
	//
	// For example: [{ "port": 80, "targetPort": 8080, "protocol": "TCP"}], shows a container port 8080 through port 80 slb exposed service, the protocol is TCP, the blank is ignored.
	Internet interface{} `field:"optional" json:"internet" yaml:"internet"`
	// Property internetSlbId: Use SLB purchased specified, currently only supports non-shared examples.
	InternetSlbId interface{} `field:"optional" json:"internetSlbId" yaml:"internetSlbId"`
	// Property intranet: Bind private SLB.
	//
	// For example: [{ "port": 80, "targetPort": 8080, "protocol": "TCP"}], shows a container port 8080 through port 80 slb exposed service, the protocol is TCP, the blank is ignored.
	Intranet interface{} `field:"optional" json:"intranet" yaml:"intranet"`
	// Property intranetSlbId: Use SLB purchased specified, currently only supports non-shared examples.
	IntranetSlbId interface{} `field:"optional" json:"intranetSlbId" yaml:"intranetSlbId"`
}

